pkg/crdexplorer/defaultexplorer/default.go

```diff
@@ -13,8 +14,9 @@ import (
 // DefaultExplorer contains all default operation explorer factory
 // for exploring common resource.
 type DefaultExplorer struct {
-	replicaHandlers   map[schema.GroupVersionKind]replicaExplorer
-	retentionHandlers map[schema.GroupVersionKind]RetentionExplorer
+	replicaHandlers map[schema.GroupVersionKind]replicaExplorer
+	// for retention explorer, we only care about Kind
```

Comment:
I don't think so. We might have different logic on different versions of the resource.

Reply:
but for pod, job and some other k8s defined resources, we do not care about group version, and it's sticky to define versions.

Reply:
We can't ensure different version's resources share the same structure, that's why we use GVK here.
Yes, we didn't care about this issue before but should take care of this from now.